lib/std/compress/deflate.zig

```diff
@@ -481,32 +483,52 @@ pub fn InflateStream(comptime ReaderType: type) type {
         }
 
         fn decode(self: *Self, h: *Huffman) !u16 {
-            // Fast path, read some bits and hope they're prefixes of some code
-            const prefix = try self.peekBits(PREFIX_LUT_BITS);
-            if (h.prefix_lut_len[prefix] != 0) {
-                self.discardBits(h.prefix_lut_len[prefix]);
-                return h.prefix_lut[prefix];
+            // Using u32 instead of u16 to reduce the number of casts needed.
+            var prefix: u32 = 0;
+
+            // Fast path, read some bits and hope they're the prefix of some code.
+            // We can't read PREFIX_LUT_BITS as we don't want to read past the
+            // deflate stream end, use an incremental approach instead.
+            var code_len = h.min_code_len;
+            while (true) {
+                _ = try self.peekBits(code_len);
+                // Small optimization win, use as many bits as possible in the
+                // table lookup.
+                prefix = self.bits & ((1 << PREFIX_LUT_BITS) - 1);
+
+                const lut_entry = &h.prefix_lut[prefix];
+                // The code is longer than PREFIX_LUT_BITS!
+                if (lut_entry.len == 0)
+                    break;
+                // If the code lenght doesn't increase we found a match.
```

Comment:
Pointing out this "lenght" only because I saw you had fixed it in another place 😮

Reply:
After a decade or so I keep on misspelling that damn word :(